security/src/main/java/uk/ac/ox/ndph/mts/security/authentication/AuthenticationService.java

```diff
@@ -0,0 +1,34 @@
+package uk.ac.ox.ndph.mts.security.authentication;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+
+/**
+ * The service which performs the authorisation flow
+ */
+@Service
+public class AuthenticationService {
```

Comment:
Same for the authentication part... this is when we authorise the request...
I think we can just add a method to AuthorisationService something like:
`public boolean authoriseWithManagedIdentity()  {
        String identity = env.getProperty(Managed_Identity)
        return identity.equals(securityContextUtil.getUserId());
    }`

Reply:
as above - if we use TF to inject the MSI value we would nee dot change to a service, I think? Otherwise we hard code property values in to property files in the security package ..which means rebuild/deploy when the MSI changes...

Reply:
as above, misunderstood the approach - changed code.